### PR TITLE
Split mongo indices on identifiers

### PR DESCRIPTION
**Pull request by Jasopaum**
## Fixes
Mongo indices were not used to find by identifier.

### CODE REVIEW

**Comment by tevariou** [comment]
https://blog.serverdensity.com/checking-if-a-document-exists-mongodb-slow-findone-vs-find/

It may be worth trying using find() instead of find_one() in the loader

https://stackoverflow.com/a/61182387/9208519

File: fhirstore/fhirstore.py [hunk 1]
```diff
@@ -91,11 +89,24 @@ def bootstrap(self, resource: Optional[str] = None, show_progress: Optional[bool
...
+                unique=True,
+                partialFilterExpression={
+                    "identifier.system": {"$exists": True},
+                    "identifier.value": {"$exists": True},
+                },
+            )
+            self.db[resource_type].create_index(
+                [
                     ("identifier.type.coding.system", ASCENDING),
                     ("identifier.type.coding.code", ASCENDING),
```
Comment (tevariou) [inline fhirstore/fhirstore.py:103]: Shouldn't this be:
```
("identifier.type.coding.0.code", ASCENDING),
("identifier.type.coding.0.system", ASCENDING),
```
?
See loader/src/reference_binder.py l32

File: fhirstore/fhirstore.py [hunk 2]
```diff
@@ -91,11 +89,24 @@ def bootstrap(self, resource: Optional[str] = None, show_progress: Optional[bool
...
+                    ("identifier.value", ASCENDING),
```
Comment (tevariou) [inline fhirstore/fhirstore.py]: Needed?

Comment (Jasopaum) [reply fhirstore/fhirstore.py]: Yes, it is! We misunderstood how identifiers with type are built but they always have a value and the "referential" can be indicated whether with system or with both type.coding.system and type.coding.code

File: fhirstore/fhirstore.py (hunk 1 above)
Comment (Jasopaum) [reply fhirstore/fhirstore.py:103]: We'll remove the `0` in river also, we don't want to only check the first identifier 🙂

**Review by simonvadee — APPROVED** [review]

**Pull request merged**